### PR TITLE
Initial Classes

### PR DESCRIPTION
**Pull request by mecha**

### CODE REVIEW

File: src/AbstractPropelSqlExtractor.php [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $migration The path to the propel migration class file.
+     *
+     * @return array An array containing two keys mapping to sub-arrays of up and down SQL queries, keyed by schema ID.
+     *               The keys of the returned array are {@link K_UP_SQL} and {@link K_DOWN_SQL}.
+     */
+    protected function _extract($migration)
+    {
+        $code = $this->_getMigrationClassCode(get_class($migration));
```
Comment (XedinUnknown) [inline src/AbstractPropelSqlExtractor.php:30]: If `$migration` is truly a path, as its param docs say, in which case it should probably be called e.g. `$path`, then the `$code` is almost guaranteed to be `null`.

File: src/AbstractPropelSqlExtractor.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    /**
+     * Normalizes the given direction string or string-like object.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $direction The direction.
+     *
+     * @return string The normalized direction.
+     */
+    protected function _normalizeDirection($direction)
```
Comment (XedinUnknown) [inline src/AbstractPropelSqlExtractor.php]: Seems to be unused in this class. Also, is implemented in `NormalizeDirectionCapableTrait`.

File: src/PropelSqlExtractor.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+     * @param string $rootDir The absolute path to the root directory.
+     */
+    protected function _setRootDir($rootDir)
+    {
+        $this->rootDir = $rootDir;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
```
Comment (XedinUnknown) [inline src/PropelSqlExtractor.php:124]: I feel that this implementation deserves some explanation of how it deduces the path, even if only a shallow overview.

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/AbstractPropelSqlExtractor.php (hunk 1 above)
Comment (mecha) [reply src/AbstractPropelSqlExtractor.php:30]: The doc is mistaken then. The method accepts the migration object. I will amend.

**Comment by XedinUnknown** [comment]
Overall, I can see that @mecha wasn't very enthusiastic while writing this, and I can see why: this is made to work for a single purpose only, which is to extract the result of a Propel migration into something by another specific migration library. As a result, separation of concerns has suffered quite a bit. Nevertheless, I will accept the PR once the other requested changes are made.

**Comment by XedinUnknown** [comment]
Please add branch alias for `develop`.

**Review by XedinUnknown — APPROVED** [review]

**Pull request merged**